torchtune/data/tokenizers/_tiktoken.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, Iterator, List
+
+from tiktoken import Encoding
+from tiktoken.load import load_tiktoken_bpe
+from torchtune.data.tokenizers._base import TokenEncoding
+
+# Constants controlling encode logic
+MAX_ENCODE_CHARS = 400_000
+MAX_NO_WHITESPACE_CHARS = 25_000
+
+
+class TikTokenEncoding(TokenEncoding):
+    """
+    A lightweight wrapper around tiktoken Encoding. This class additionally handles
+    breaking up the input text into substrings of a max length and splitting up long
+    repetitions to improve encode speed.
+
+    The core tiktoken.Encoding does NOT natively handle BOS and EOS ids, so this class
+    does not either. They must be handled by the caller of ``encode``.
+
+    Args:
+        path (str): Path to pretrained tokenizer checkpoint file.
+        name (str): Name of the tokenizer (used by tiktoken for identification).
+        pattern (str): Regex pattern used to split input text into chunks before passing
+            to byte-pair encoding.
+        special_tokens (Dict[str, int]): Mapping of special tokens to their ids.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        name: str,
+        pattern: str,
+        special_tokens: Dict[str, int],
+    ):
+        mergeable_ranks = load_tiktoken_bpe(path)
+        self.tt_model = Encoding(
+            name=name,
+            pat_str=pattern,
+            mergeable_ranks=mergeable_ranks,
+            special_tokens=special_tokens,
+        )
+        # Vocab size without special tokens
+        self.base_vocab_size = len(mergeable_ranks)
+        # Vocab size with special tokens
+        self.vocab_size = self.tt_model.n_vocab
+
+    def _split_long_repetitions(
```

Comment:
kind of a nit but strictly speaking this does not need to be a method on the class, right?

Reply:
this is only specific to tiktoken and will not be used elsewhere, correct? but I don't feel strongly either way